crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -238,13 +257,58 @@ impl Consensus {
             ),
         }
     }
-    #[allow(unused)]
+    pub fn validate_block_time(
+        block_timestamp: u32,
+        mtp: u32,
+        time: impl NodeTime,
+    ) -> Result<(), BlockValidationErrors> {
+        if mtp > block_timestamp && block_timestamp > (time.get_time().sub(2 * HOUR)) {
```

Comment:
Don't inline all those statements together.  For example, for the timestamp part you can do: `let now = time.get_time()` and `let max = time.get_time() + 2 * HOUR`